include/matx/executors/stf.h

```diff
@@ -0,0 +1,253 @@
+          cuda::experimental::stf::async_resources_handle handle;
+          ctx_ = cuda::experimental::stf::stream_ctx(reinterpret_cast<cudaStream_t>(stream), handle);
+          //ctx_ = cuda::experimental::stf::graph_ctx(reinterpret_cast<cudaStream_t>(stream), handle);
+      }
+
+      /**
+       * @brief Construct a new stfExecutor object using the default stream
+       * 
+       */
+      stfExecutor() : stream_(0) {
+          ctx_ = cuda::experimental::stf::stream_ctx();
+          //ctx_ = cuda::experimental::stf::graph_ctx();
+      }
+
+      /**
+       * @brief Returns stream associated with executor
+      */
+      auto getStream() const { return stream_; }
+
+      /**
+       * @brief Get CUDASTF Ctx
+       * 
+       */
+      auto &getCtx() const noexcept { return ctx_; }
+
+      /**
+       * @brief Synchronize the cuda executor's stream
+       * 
+       */
+      void sync() { cudaStreamSynchronize(stream_); }
+
+      /**
+       * Execute an operator on a device
+       * 
+       * @tparam Op Operator type
+       * @param op value
+       **/
+      template <typename Op>
+        void Exec(Op &op) const {
+            //std::cout << "exec on stfexecutor -- start\n";
+#ifdef __CUDACC__      
+          dim3 threads, blocks;  
+
+          auto ctx = getCtx();
+          // Parameters passed by value in CUDA are limited to 4096B. If the user exceeds this, we 
+          // need to error out and have them break up the statement
+          MATX_STATIC_ASSERT((sizeof(op) + sizeof(index_t) * Op::Rank()) <= CUDA_MAX_VAL_PARAM, 
+              "Parameter buffer to device is limited to 4096B. Please break up your operator statement into multiple executions to limit the size of the parameters");
+
+          if constexpr (Op::Rank() == 0) {
+            threads = 1;
+            blocks = 1;
+            if constexpr (is_matx_op_lvalue<Op>() || is_matx_set_op<Op>()) {
+                    auto tsk = ctx.task();
+                    tsk.set_symbol(op.str());
+                    op.apply_dep_to_task(tsk); // recursively find the tensors from the tree to apply deps
+                    //std::cout << "Start launch task. Rank = " << Op::Rank() << " " << op.str() << '\n';
+                    tsk->*[&](cudaStream_t s) { 
+                        detail::matxOpT0Kernel<<<blocks, threads, 0, s>>>(op);
+                    };
+                    //std::cout << "End launch task.\n";
+            }
+            else {
+                //std::cout << " RANK 0 not on LHS operator = " << op.str() << '\n';
+                detail::matxOpT0Kernel<<<blocks, threads, 0, stream_>>>(op);
+            }
+          }
+          else {
+            cuda::std::array<index_t, Op::Rank()> sizes;
+            for (int i = 0; i < Op::Rank(); i++) {
+              sizes[i] = op.Size(i);
+            }        
+
+            bool stride = detail::get_grid_dims<Op::Rank()>(blocks, threads, sizes, 256);
+
+            if constexpr (Op::Rank() == 1) {
```

Comment:
It looks like we could factorize all that constexpr cascade, and move the constexpr tests into the lambda ?